openstack/compute/v2/extensions/keypairs/requests.go

```diff
@@ -76,14 +80,28 @@ func Create(client *gophercloud.ServiceClient, opts CreateOptsBuilder) (r Create
 
 // Get returns public data about a previously uploaded KeyPair.
 func Get(client *gophercloud.ServiceClient, name string) (r GetResult) {
-	resp, err := client.Get(getURL(client, name), &r.Body, nil)
+	r = GetWithUserID(client, name, "")
+	return
+}
+
+// GetWithUserID returns public data about a previously uploaded KeyPair, owned by a specific userID.
+// If the userID is an empty string, it searches the KeyPairs of the current user.
+func GetWithUserID(client *gophercloud.ServiceClient, name string, userID string) (r GetResult) {
+	resp, err := client.Get(getURL(client, name, userID), &r.Body, nil)
 	_, r.Header, r.Err = gophercloud.ParseResponse(resp, err)
 	return
 }
 
 // Delete requests the deletion of a previous stored KeyPair from the server.
 func Delete(client *gophercloud.ServiceClient, name string) (r DeleteResult) {
-	resp, err := client.Delete(deleteURL(client, name), nil)
+	r = DeleteWithUserID(client, name, "")
+	return
+}
+
+// DeleteWithUserID requests the deletion of a previous stored KeyPair from the server, owned by a specific userID.
+// If the userID is an empty string, it deletes the KeyPair of the calling user.
+func DeleteWithUserID(client *gophercloud.ServiceClient, name string, userID string) (r DeleteResult) {
```

Comment:
Similar here: there needs to be a `DeleteOpts` struct created: https://github.com/gophercloud/gophercloud/blob/master/openstack/blockstorage/v3/qos/requests.go#L67-L100